flash/vision/classification/data.py

```diff
@@ -13,369 +13,392 @@
+
+        return num_classes
+
+    @classmethod
+    def instantiate_preprocess(
+        cls,
+        train_transform: Dict[str, Union[nn.Module, Callable]],
+        valid_transform: Dict[str, Union[nn.Module, Callable]],
+        test_transform: Dict[str, Union[nn.Module, Callable]],
+        predict_transform: Dict[str, Union[nn.Module, Callable]],
+        preprocess_cls: Type[Preprocess] = None
+    ) -> Preprocess:
+
+        train_transform, valid_transform, test_transform, predict_transform = cls._resolve_transforms(
+            train_transform, valid_transform, test_transform, predict_transform
         )
 
-        return cls(
-            train_ds=train_ds,
-            valid_ds=valid_ds,
-            test_ds=test_ds,
-            batch_size=batch_size,
-            num_workers=num_workers,
+        preprocess_cls = preprocess_cls or cls.preprocess_cls
+        return preprocess_cls(train_transform, valid_transform, test_transform, predict_transform)
+
+    @classmethod
+    def _resolve_transforms(
+        cls,
+        train_transform: Optional[Union[str, Dict]] = 'default',
+        valid_transform: Optional[Union[str, Dict]] = 'default',
+        test_transform: Optional[Union[str, Dict]] = 'default',
+        predict_transform: Optional[Union[str, Dict]] = 'default',
+    ):
+
+        if isinstance(train_transform, str) and train_transform == 'default':
+            train_transform = cls.default_train_transforms()
+
+        if isinstance(valid_transform, str) and valid_transform == 'default':
+            valid_transform = cls.default_valid_transforms()
+
+        if isinstance(test_transform, str) and test_transform == 'default':
+            test_transform = cls.default_valid_transforms()
+
+        if isinstance(predict_transform, str) and predict_transform == 'default':
+            predict_transform = cls.default_valid_transforms()
+
+        return (
+            cls._check_transforms(train_transform), cls._check_transforms(valid_transform),
+            cls._check_transforms(test_transform), cls._check_transforms(predict_transform)
         )
 
     @classmethod
     def from_folders(
         cls,
-        train_folder: Optional[Union[str, pathlib.Path]],
-        train_transform: Optional[Callable] = _DEFAULT_TRAIN_TRANSFORMS,
+        train_folder: Optional[Union[str, pathlib.Path]] = None,
         valid_folder: Optional[Union[str, pathlib.Path]] = None,
-        valid_transform: Optional[Callable] = _DEFAULT_VALID_TRANSFORMS,
         test_folder: Optional[Union[str, pathlib.Path]] = None,
-        loader: Callable = _pil_loader,
+        predict_folder: Union[str, pathlib.Path] = None,
+        train_transform: Optional[Union[str, Dict]] = 'default',
+        valid_transform: Optional[Union[str, Dict]] = 'default',
+        test_transform: Optional[Union[str, Dict]] = 'default',
+        predict_transform: Optional[Union[str, Dict]] = 'default',
```

Comment:
I feel that string 'default' is quite a big source of the issue with an eventual typo, can we rather use None?

Reply:
I agree. In another PR.